Snowflake.Data/Configuration/EasyLoggingConfigFinder.cs

```diff
@@ -76,6 +91,35 @@ private string SearchForConfigInDirectory(Func<string> directoryProvider, string
             }
         }
 
-        private string OnlyIfFileExists(string filePath) => _fileOperations.Exists(filePath) ? filePath : null;
+        private string OnlyIfFileExists(string filePath, string directoryDescription)
+        {
+            if (_fileOperations.Exists(filePath))
+            {
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    CheckIfValidPermissions(filePath);
+                }
+                s_logger.Info($"Using config file specified from {directoryDescription} directory");
+                return filePath;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private void CheckIfValidPermissions(string filePath)
+        {
+            // Check if others have permissions to modify the file and fail if so
+            int filePermissions;
+            string commandParameters = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? "-c '%a'" : "-f %A";
+            bool isParsed = int.TryParse(EasyLoggerUtil.CallBash($"stat {commandParameters} {filePath}"), out filePermissions);
```

Comment:
Ok I'll check out that library

Reply:
I've tried using `Mono.Unix` and it seems to be working. The permissions in `Mono.Unix.FileAccessPermissions` are in hexadecimal so I had to convert it so it can be compared to umask. If you know of a better way to do that conversion let me know